irksome/tools.py

```diff
@@ -1,11 +1,13 @@
-from firedrake import MixedVectorSpaceBasis
-from ufl.algorithms.analysis import has_exact_type
+import numpy
+from firedrake import MixedVectorSpaceBasis, split
+from ufl.algorithms.analysis import extract_type, has_exact_type
 from ufl.algorithms.map_integrands import map_integrand_dags
 from ufl.classes import CoefficientDerivative
 from ufl.constantvalue import as_ufl
 from ufl.corealg.multifunction import MultiFunction
 from ufl.log import error
-import numpy
+
```

Comment:
Blank line in imports?

Reply:
Easy to fix.  I'm surprised that's not a flake8 rule...